src/api_midlevel.jl

```diff
@@ -0,0 +1,46 @@
+# This file defines midlevel api wrappers. We include name normalization for methods that are
+# applicable to different hdf5 api-layers. We still try to adhere close proximity to the underlying
+# method name in the hdf5-library.
+
+"""
+    HDF5.set_extent_dims(dset::HDF5.Dataset, new_dims::Dims)
+
+Change the current dimensions of a dataset to `new_dims`, limited by
+`max_dims = get_simple_extent_dims(dset)[2]`. Reduction is possible and leads to loss of truncated data.
+"""
+function set_extent_dims(dset::Dataset, size::Dims)
+    checkvalid(dset)
+    h5d_set_extent(dset, hsize_t[reverse(size)...])
+end
+
+"""
+    HDF5.set_extent_dims(dset::HDF5.Dataspace, new_dims::Dims, max_dims::Union{Dims,Nothing} = nothing)
+
+Change the current dimensions of a dataspace `dset` to `new_dims`, limited by
+`max_dims = get_simple_extent_dims(dset)[2]`. Reduction is possible and leads to loss of truncated data.
+"""

```

Comment:
```suggestion
"""
    HDF5.set_extent_dims(dset::HDF5.Dataspace, new_dims::Dims, max_dims::Union{Dims,Nothing} = nothing)

Change the dimensions of a dataspace `dspace` to `new_dims`, optionally with the maximum possible
dimensions `max_dims` different from the active size `new_dims`. If not given, `max_dims` is set equal
to `new_dims`.
"""
```

Two sets of changes — 1) simple rename `dset` -> `dspace` in this doc string. 2) I think this doc string needs to explain `max_dims` (especially what the default `nothing` means), so I've suggested a sample rewrite (but feel free to reword yet again).